### PR TITLE
Scene cleanups

### PR DESCRIPTION
**Pull request by louist103**
u32 segment -> void* and add symbols to sEntranceCutsceneTable

### CODE REVIEW

File: assets/xml/scenes/dungeons/ddan.xml [hunk 1]
```diff
@@ -1,5 +1,18 @@
 <Root>
     <File Name="ddan_scene" Segment="2">
+        <Texture Name="gDcDayEntranceTex" OutName="day_entrance" Format="ci8" Width="32" Height="64" Offset="0x12378" TlutOffset="0x11D70"/>
+        <Texture Name="gDcNightEntranceTex" OutName="night_entrance" Format="ci8" Width="32" Height="64" Offset="0x13378" TlutOffset="0x11D70"/>
```
Comment (fig02) [inline assets/xml/scenes/dungeons/ddan.xml:4]: DC being an acronym should be capitalized I think

File: assets/xml/scenes/dungeons/men.xml [hunk 2]
```diff
@@ -1,5 +1,7 @@
 <Root>
     <File Name="men_scene" Segment="2">
+        <Texture Name="gGtgDayEntranceTex" OutName="day_entrance" Format="ia16" Width="8" Height="128" Offset="0xF8C0"/>
+        <Texture Name="gGtgNightEntranceTex" OutName="night_entrance" Format="ia16" Width="8" Height="128" Offset="0x100C0"/>
```
Comment (fig02) [inline assets/xml/scenes/dungeons/men.xml:4]: same with GTG

File: include/variables.h [hunk 3]
```diff
@@ -2932,7 +2932,6 @@ extern GameInfo* gGameInfo; // 0x8015FA90
 extern u16 D_8015FCC0;
 extern u16 D_8015FCC2;
 extern u16 D_8015FCC4;
-//extern ? D_8015FCC6;
 extern u8 D_8015FCC8;
 //extern ? D_8015FCCA;
```
Comment (fig02) [inline include/variables.h:2936]: can probably remove the other too?

File: src/code/z_scene_table.c [hunk 4]
```diff
@@ -995,8 +1013,8 @@ void func_80099760(GlobalContext* globalCtx) {
     CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_scene_table.c", 4859);
 }
 
-u32 D_8012A300[] = { 0x02012378, 0x02013378 };
-u32 D_8012A308[] = { 0x02011F78, 0x02014778, 0x02014378, 0x02013F78, 0x02014B78, 0x02013B78, 0x02012F78, 0x02012B78 };
+void* gDcEntranceTextures[] = { gDcDayEntranceTex, gDcNightEntranceTex };
+void* sDcLaveFloorTextures[] = { gDcLavaFloor1Tex, gDcLavaFloor2Tex, gDcLavaFloor3Tex, gDcLavaFloor4Tex, gDcLavaFloor5Tex, gDcLavaFloor6Tex, gDcLavaFloor7Tex, gDcLavaFloor8Tex };
```
Comment (fig02) [inline src/code/z_scene_table.c:1017]: Lave -> Lava

File: include/variables.h (hunk 3 above)
Comment (fig02) [reply include/variables.h:2936]: its local to z_demo though, only global things need to go here

File: assets/xml/scenes/dungeons/MIZUsin.xml [hunk 5]
```diff
@@ -1,5 +1,7 @@
 <Root>
     <File Name="MIZUsin_scene" Segment="2">
+        <Texture Name="gWaterTempleDayEntranceTex" OutName="day_entrance" Format="ia16" Width="8" Height="64" Offset="0x14C30"/>
+        <Texture Name="gWaterTempleNightEntranceTex" OutName="night+entrance" Format="ia16" Width="8" Height="64" Offset="0x15830"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/dungeons/MIZUsin.xml:4]:
```suggestion
        <Texture Name="gWaterTempleNightEntranceTex" OutName="night_entrance" Format="ia16" Width="8" Height="64" Offset="0x15830"/>
```

File: assets/xml/scenes/dungeons/ddan.xml [hunk 6]
```diff
@@ -1,5 +1,18 @@
...
+
+        <!-- these need to have TLUTs extract from room 0 at 0x11290-->
+        <Texture Name="gDcLavaFloor1Tex" OutName="lave_floor_1" Format="ci8" Width="32" Height="32" Offset="0x11F78" />
+        <Texture Name="gDcLavaFloor2Tex" OutName="lave_floor_2" Format="ci8" Width="32" Height="32" Offset="0x14778" />
+        <Texture Name="gDcLavaFloor3Tex" OutName="lave_floor_3" Format="ci8" Width="32" Height="32" Offset="0x14378" />
+        <Texture Name="gDcLavaFloor4Tex" OutName="lave_floor_4" Format="ci8" Width="32" Height="32" Offset="0x13F78" />
+        <Texture Name="gDcLavaFloor5Tex" OutName="lave_floor_5" Format="ci8" Width="32" Height="32" Offset="0x14B78" />
+        <Texture Name="gDcLavaFloor6Tex" OutName="lave_floor_6" Format="ci8" Width="32" Height="32" Offset="0x13B78" />
+        <Texture Name="gDcLavaFloor7Tex" OutName="lave_floor_7" Format="ci8" Width="32" Height="32" Offset="0x12F78" />
+        <Texture Name="gDcLavaFloor8Tex" OutName="lave_floor_8" Format="ci8" Width="32" Height="32" Offset="0x12B78" />
```
Comment (Dragorn421) [inline assets/xml/scenes/dungeons/ddan.xml:14]:
```suggestion
        <Texture Name="gDcLavaFloor8Tex" OutName="lava_floor_8" Format="ci8" Width="32" Height="32" Offset="0x12B78" />
```
and the others too

File: assets/xml/scenes/dungeons/jyasinboss.xml [hunk 7]
```diff
@@ -1,5 +1,6 @@
 <Root>
     <File Name="jyasinboss_scene" Segment="2">
+        <Cutscene Name="gNabooruKnuckleIntroCs" Offset="0x2BB0"/>
         <Cutscene Name="gNabooruKnuckleDefeatCs" Offset="0x3F80"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/dungeons/jyasinboss.xml:4]: should this also have a prefix like `SpiritBoss` or something (related to the scene)

File: assets/xml/scenes/dungeons/ydan.xml [hunk 8]
```diff
@@ -1,5 +1,9 @@
-<Root>
+<Root> <!--gSpot03Path_006918-->
```
Comment (Dragorn421) [inline assets/xml/scenes/dungeons/ydan.xml:1]: what is this comment for

File: assets/xml/scenes/indoors/tokinoma.xml [hunk 9]
```diff
@@ -2,6 +2,7 @@
     <File Name="tokinoma_scene" Segment="2">
         <Cutscene  Name="gFirstAdultCs" Offset="0x46F0"/>
         <Cutscene  Name="gPreludeCs" Offset="0x6D20"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/indoors/tokinoma.xml:4]: not your change but since you're introducing scene prefixes, maybe those should use them as well

File: assets/xml/scenes/misc/hakaana_ouke.xml [hunk 10]
```diff
@@ -1,6 +1,7 @@
 <Root>
     <File Name="hakaana_ouke_scene" Segment="2">
         <Cutscene Name="gSunSongTeachCs" Offset="0x24A0"/>
+        <Cutscene Name="gSunSongTeachPart2Cs" Offset="0x28E0"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/misc/hakaana_ouke.xml:4]: scene prefix?

File: assets/xml/scenes/overworld/spot00.xml [hunk 11]
```diff
@@ -1,6 +1,12 @@
 <Root>
     <File Name="spot00_scene" Segment="2">
+        <Cutscene Name="gHyruleFieldGetOotCs" Offset="0xBB80"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/overworld/spot00.xml:3]:
```suggestion
        <Cutscene Name="gHyruleFieldGetOoTCs" Offset="0xBB80"/>
```

File: assets/xml/scenes/overworld/spot00.xml [hunk 12]
```diff
@@ -1,6 +1,12 @@
...
         <Cutscene Name="gZeldaSongOfTimeCs" Offset="0xF870"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/overworld/spot00.xml:4]:
```suggestion
        <Cutscene Name="gHyruleFieldZeldaSongOfTimeCs" Offset="0xF870"/>
```

File: assets/xml/scenes/overworld/spot02.xml [hunk 13]
```diff
@@ -6,6 +6,7 @@
         <Cutscene Name="spot02_sceneCutsceneData0x003C80" Offset="0x3C80"/>
 
         <Cutscene Name="spot02_scene_Cs_005020" Offset="0x5020"/>
+        <Cutscene Name="gGraveYardIntroCs" Offset="0x70C0"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/overworld/spot02.xml:9]:
```suggestion
        <Cutscene Name="gGraveyardIntroCs" Offset="0x70C0"/>
```
it's a single word

File: assets/xml/scenes/overworld/spot04.xml [hunk 14]
```diff
@@ -3,7 +3,7 @@
         <Path Name="gSpot04Path_000314" Offset="0x314"/>
         <Path Name="gSpot04Path_00031C" Offset="0x31C"/>
         <Path Name="gSpot04Path_00D730" Offset="0xD730"/>
-
+        <Cutscene Name="gDekuSproutCs" Offset="0xC9D0"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/overworld/spot04.xml:6]:
```suggestion
        <Cutscene Name="gKokiriForestDekuSproutCs" Offset="0xC9D0"/>
```

File: assets/xml/scenes/overworld/spot06.xml [hunk 15]
```diff
@@ -3,6 +3,7 @@
         <Cutscene Name="gFireArrowsCS" Offset="0x7020"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/overworld/spot06.xml:3]:
```suggestion
        <Cutscene Name="gLakeHyliaFireArrowsCS" Offset="0x7020"/>
```

File: assets/xml/scenes/overworld/spot12.xml [hunk 16]
```diff
@@ -1,7 +1,9 @@
 <Root>
     <File Name="spot12_scene" Segment="2">
-        <Cutscene Name="gSpot12Cs_006490" Offset="0x6490"/>
-
+        <Cutscene Name="gGerudoValleyFirstCaptureCs" Offset="0x55C0"/>
+        <Cutscene Name="gGerudoFortressIntroCs" Offset="0x6490"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/overworld/spot12.xml:4]: GerudoValley or GerudoFortress

File: assets/xml/scenes/overworld/spot16.xml [hunk 17]
```diff
@@ -1,6 +1,7 @@
 <Root>
     <File Name="spot16_scene" Segment="2">
         <Cutscene Name="gDmtOwlCs" Offset="0x1E6A0"/>
+        <Cutscene Name="gDmtIntroCs" Offset="0x7EA0"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/overworld/spot16.xml:4]: acronym -> DMT

File: assets/xml/scenes/overworld/spot17.xml [hunk 18]
```diff
@@ -1,6 +1,7 @@
 <Root>
     <File Name="spot17_scene" Segment="2">
         <Cutscene Name="gBoleroCs" Offset="0x45D0"/>
```
Comment (Dragorn421) [inline assets/xml/scenes/overworld/spot17.xml:3]:
```suggestion
        <Cutscene Name="gDeathMountainCraterBoleroCs" Offset="0x45D0"/>
```

File: src/code/z_scene_table.c [hunk 19]
```diff
@@ -1712,8 +1730,8 @@ void func_8009D31C(GlobalContext* globalCtx) {
     CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_scene_table.c", 6528);
 }
 
-u32 D_8012A350[] = { 0x02006550, 0x02003550 };
-u32 D_8012A358[] = { 0x02002350, 0x02001350 };
+void* sGuardHouseView2Texures[] = { gGuardHouseOutSideView1DayTex, gGuardHouseOutSideView1NightTex };
```
Comment (Dragorn421) [inline src/code/z_scene_table.c:1733]:
```suggestion
void* sGuardHouseView2Textures[] = { gGuardHouseOutSideView1DayTex, gGuardHouseOutSideView1NightTex };
```

File: src/code/z_scene_table.c [hunk 20]
```diff
@@ -2300,8 +2318,8 @@ void func_8009FC90(GlobalContext* globalCtx) {
     CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_scene_table.c", 7653);
 }
 
-f32 D_8012A398 = 0.0f;
 
+f32 D_8012A398 = 0.0f;
 void func_8009FE58(GlobalContext* globalCtx) {
```
Comment (Dragorn421) [inline src/code/z_scene_table.c:2323]: unintended change? I don't see the point

File: include/variables.h (hunk 3 above)
Comment (louist103) [reply include/variables.h:2936]: `D_8015FCC0` Used in treemouth.c
`D_8015FCC2` Used in treemouth.c
`D_8015FCC4` Used in treemouth.c
`D_8015FCC8` Used in dbcamera.c and kankyo.c

Comment (fig02) [reply include/variables.h:2936]: I was talking about D_8015FCCA, by "other" i meant the other which is commented out

Comment (louist103) [reply include/variables.h:2936]: Ok I wasn't sure.  I took it out anyway but I guess it came back.

File: src/code/z_demo.c [hunk 21]
```diff
@@ -24,31 +48,57 @@ CutsceneStateHandler sCsStateHandlers2[] = {
...
+    { 0x054C, AGE_REQ_NONE, 0xAD, gSpiritBarrierCs },
+    { 0x008D, AGE_REQ_ADULT, 0xC0, gSpiritBossNabooruKnuckleIntroCs },
+    { 0x03B4, AGE_REQ_ADULT, 0xC7, gGerudoFortressFirstCaptureCs },
+    { 0x0246, AGE_REQ_NONE, 0xB9, gDeathMountainCraterIntroCs },
+    { 0x05E8, AGE_REQ_NONE, 0xC6, gKokiriForestDekuSproutCs },
 };
 
 // Unused, seems to be an early list of dungeon entrance cutscene locations
-u32 D_8011E304[] = { 0x0200B650, 0x02015600, 0x02014F80, 0x02003F80, 0x02000330, gTowerBarrierCs };
+void* D_8011E304[] = { 0x0200B650, 0x02015600, 0x02014F80, 0x02003F80, 0x02000330, gTowerBarrierCs };
```
Comment (fig02) [inline src/code/z_demo.c:94]: if this is unused do we really know that this symbol belongs to these addresses? without context of what scenes this is used under, we cant really tell right

Comment (louist103) [reply src/code/z_demo.c:94]: Probably not no. But I just decided to leave it there since it was already set.  I can take it out though.

Comment (fig02) [reply src/code/z_demo.c:94]: oh, yeah i didnt realize this wasnt changed by you. I pose that question to others then, and wouldnt mind removing it personally

File: src/code/z_scene_table.c [hunk 22]
```diff
@@ -995,8 +1013,8 @@ void func_80099760(GlobalContext* globalCtx) {
     CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_scene_table.c", 4859);
 }
 
-u32 D_8012A300[] = { 0x02012378, 0x02013378 };
-u32 D_8012A308[] = { 0x02011F78, 0x02014778, 0x02014378, 0x02013F78, 0x02014B78, 0x02013B78, 0x02012F78, 0x02012B78 };
+void* gDCEntranceTextures[] = { gDCDayEntranceTex, gDCNightEntranceTex };
+void* sDCLavaFloorTextures[] = { gDCLavaFloor1Tex, gDCLavaFloor2Tex, gDCLavaFloor3Tex, gDCLavaFloor4Tex, gDCLavaFloor5Tex, gDCLavaFloor6Tex, gDCLavaFloor7Tex, gDCLavaFloor8Tex };
```
Comment (Roman971) [inline src/code/z_scene_table.c:1017]: This looks like it needs to be re-formatted, and I suggest adding trailing commas to all these texture pointer arrays to improve the formatting and make it consistent.

File: src/code/z_scene_table.c [hunk 23]
```diff
@@ -2300,7 +2318,7 @@ void func_8009FC90(GlobalContext* globalCtx) {
     CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_scene_table.c", 7653);
 }
 
-f32 D_8012A398 = 0.0f;
+f32 D_8012A398 = 0.0f; //Can not be in function static
```
Comment (Roman971) [inline src/code/z_scene_table.c:2321]: Imo we don't need to have comments for this type of information

File: src/code/z_demo.c [hunk 24]
```diff
@@ -24,31 +51,59 @@ CutsceneStateHandler sCsStateHandlers2[] = {
...
-u32 D_8011E304[] = { 0x0200B650, 0x02015600, 0x02014F80, 0x02003F80, 0x02000330, gTowerBarrierCs };
+void* D_8011E304[] = {
+    gDekuTreeIntroCs, gJabuJabuIntroCs, gDcOpeningCs, gMinuetCs, gIceCavernSerenadeCs, gTowerBarrierCs,
+};
 
 u16 D_8015FCC0;
 u16 D_8015FCC2;
 u16 D_8015FCC4;
-s16 D_8015FCC6;
+s16 sActiveCam;
```
Comment (Roman971) [inline src/code/z_demo.c:104]: A similar change is already being done in #989 and is still under discussion, so it would probably be better to leave it out of this PR

File: include/z64cutscene.h [hunk 25]
```diff
@@ -7,7 +7,7 @@ typedef struct {
     /* 0x00 */ u16 entrance;       // entrance index upon which the cutscene should trigger
     /* 0x02 */ u8  ageRestriction; // 0 for adult only, 1 for child only, 2 for both ages
     /* 0x03 */ u8  flag;           // eventChkInf flag bound to the entrance cutscene
-    /* 0x04 */ u32 segAddr;        // segment offset location of the cutscene
+    /* 0x04 */ void* segAddr;        // segment offset location of the cutscene
```
Comment (Roman971) [inline include/z64cutscene.h:10]:
```suggestion
    /* 0x04 */ void* segAddr;      // segment offset location of the cutscene
```

File: assets/xml/scenes/overworld/spot15.xml [hunk 26]
```diff
@@ -1,5 +1,7 @@
 <Root>
     <File Name="spot15_scene" Segment="2">
+        <Cutscene Name="gHyruleCastleIntroCs" Offset="0x3F40"/>
+        <!--<Cutscene Name="gHyruleCastleIntro2Cs" Offset="0x4280"/> There is an entry in the Entrance table but its not a valid cutscene -->
```
Comment (Roman971) [inline assets/xml/scenes/overworld/spot15.xml:4]: What is this referring to? I don't see any corresponding entry in the table but I might be looking at the wrong thing

File: src/code/z_demo.c [hunk 27]
```diff
@@ -24,31 +51,59 @@ CutsceneStateHandler sCsStateHandlers2[] = {
 
 u8 sTitleCsState = 0;
 
-// TODO: use symbols for these cutscene locations once scenes and rooms are in C
+// Todo use enum when one is created
+#define AGE_REQ_CHILD 1
+#define AGE_REQ_ADULT 0
+#define AGE_REQ_NONE 2
```
Comment (Roman971) [inline src/code/z_demo.c:57]: Imo this doesn't really belong in this PR or at least warrants more discussion.

I personally don't think we should use any special defines here since the values actually correspond to `linkAge`, except `2` which is used as an arbitrary value and might warrant a define but I'm not sure it's even worth one tbh

**Review by Roman971 — APPROVED** [review]
